src/Core/Config/Capability/ISetConfigValuesTransactional.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Core\Config\Capability;
+
+use Friendica\Core\Config\Exception\ConfigPersistenceException;
+
+/**
+ * Interface for transactional saving of config values
+ * It buffers every set/delete until "save()" is called
+ */
+interface ISetConfigValuesTransactional
+{
+	/**
+	 * Get a particular user's config variable given the category name
+	 * ($cat) and a $key.
+	 *
+	 * Get a particular config value from the given category ($cat)
+	 *
+	 * @param string  $cat        The category of the configuration value
+	 * @param string  $key           The configuration key to query
+	 *
+	 * @return mixed Stored value or null if it does not exist
+	 *
+	 * @throws ConfigPersistenceException In case the persistence layer throws errors
+	 *
+	 */
+	public function get(string $cat, string $key);
```

Comment:
Why do we need a `get` function in the config transaction named "I**Set**ConfigValues"? Do you call the `get` method anywhere else than the tests?

Reply:
Not yet. but it's helpful in cases where only a `TransactionConfig` is available and to avoid that someone thinks he can `get()` transaction config changes per "normal" `Config` class. So I thought why not adding it here :)

Reply:
I feel like it makes it confusing, that's another `Config` object with `->get` `->set` and `->delete`. The interface is clear about the use case, I don't think it should have a `get` method. The object itself may define it, but I believe it shouldn't be in the interface.